src/main/java/org/jabref/gui/externalfiles/ImportHandler.java

```diff
@@ -194,52 +194,96 @@ public void importEntries(List<BibEntry> entries) {
     }
 
     public void importEntryWithDuplicateCheck(BibDatabaseContext bibDatabaseContext, BibEntry entry) {
-        ImportCleanup cleanup = new ImportCleanup(bibDatabaseContext.getMode());
-        BibEntry cleanedEntry = cleanup.doPostCleanup(entry);
+        BibEntry cleanedEntry = cleanUpEntry(bibDatabaseContext, entry);
+
+        Optional<BibEntry> existingDuplicateInLibrary = findDuplicate(bibDatabaseContext, cleanedEntry);
         BibEntry entryToInsert = cleanedEntry;
```

Comment:
I would move this two lines up, because this line has no relation to line 199, but to line 197 (because it stores that value in another variable)